pilot/pkg/model/gateway.go

```diff
@@ -49,7 +49,7 @@ func MergeGateways(gateways ...Config) *MergedGateway {
 	log.Debugf("MergeGateways: merging %d gateways", len(gateways))
 	for _, spec := range gateways {
 		name := ResolveShortnameToFQDN(spec.Name, spec.ConfigMeta)
-		names[name.String()] = true
+		names[string(name)] = true
```

Comment:
Oh, that's a bummer. When I wrote `String()` I assumed the compiler would inline it since it's only a type assertion.

Reply:
yeah, this incurred a _bit_ of a penalty when looking at pprof. Took it out for good measure.